### PR TITLE
Explorer by Kazoobard

### PR DESCRIPTION
**Pull request by KazooBard**
Adds the Explorer, a new cargonian whose purpose is to explore, loot and mine space rocks for cargonia!

## About The Pull Request

This adds the Explorer, a cargonian whose equipment was so expensive, he does not even get his own room. His equipment also got slightly reduced in quality, such as the poor man's RCD that has only 50 storage units instead of the regular 160. The explorer's job is to, well, explore space and the gateway, bringing cargonia loot and ore. The explorer starts with a beacon allowing him to choose one of the three sets of equipment, being the medical specialization, self defense specialization and engineering specialization. 
The explorer's gear is similar to the miner, but he had nanotrens put it on him, so that they make only as much knives, ore bags and mining scanners as it is needed.

The jungle map
## Why It's Good For The Game
I had the code ready and there needs to be a dedicated job for exploring space and the gateway, since assistants breaking is is not a good way to do this. HoPs will be spared a headache. Additional miners. And it is fun.

## Changelog
:cl:
add: Added Explorer job
add: Added Explorer Action figure
add: Added exploring equipment beacon
add: Added landmarks for the explorer to start in cargo's abandoned storage
add: Added Poor man's RCD
/:cl:

### CODE REVIEW

**Comment by ghost** [comment]
I don't understand the purpose of the Explorer, they're basically just a worse miner?
The duty of space exploration already falls to the Curator and the Shaft Miners, aside from that everyone else can do it too, and the gateway is not enabled on this server, so this really just is a worse Miner in its current state.

**Comment by KazooBard** [comment]
@BeloneX the purpose is not only to dig, but to explore and bring back loot and valuables previously unobtainable by only doing mining. It exists to encourage space exploration as an actual job instead of a gimmick and to bring the gateway back to life.

**Comment by ghost** [comment]
I can somewhat see your point there, so for now my stance is going to be neutral on this pull request.

**Comment by utilbird** [comment]
isn't miner just a better version of this job after they've gotten their lavaland loot?

**Comment by KazooBard** [comment]
> isn't miner just a better version of this job after they've gotten their lavaland loot?
No. The space loot is different and has different purposes: example, hilton hotel is a lot of fun, pocket dimension to hide in and heal. Laser guns, gateway syndicate mechs at your disposal, more guns and fun stuff. This is not a challenge from a single mob, but a challenge from the enviroment, as space is very deadly.

File: code/game/objects/items/RCD.dm [hunk 1]
```diff
@@ -953,3 +953,8 @@ RLD
 #undef GLOW_MODE
 #undef LIGHT_MODE
 #undef REMOVE_MODE
+
+/obj/item/construction/rcd/poor
+	name = "Smoking RCD"
```
Comment (Archanial) [inline code/game/objects/items/RCD.dm:958]:
```suggestion
	name = "Damaged RCD"	
```
sounds better than smoking

File: code/game/objects/items/miscellaneous.dm [hunk 2]
```diff
@@ -144,6 +144,67 @@
 	new choice(get_turf(M))
 	to_chat(M, "You hear something crackle from the beacon for a moment before a voice speaks.  \"Please stand by for a message from S.E.L.F. Message as follows: <span class='bold'>Item request received. Your package has been transported, use the autosurgeon supplied to apply the upgrade.</span> Message ends.\"")
 
+/obj/item/choice_beacon/exploring
+	name = "Exploring equipment delivery beacon"
+	desc = "A device to order exploring equipment"
```
Comment (Archanial) [inline code/game/objects/items/miscellaneous.dm:149]:
```suggestion
	desc = "A device used to order exploring equipment"
```

File: code/game/objects/items/miscellaneous.dm [hunk 3]
```diff
@@ -144,6 +144,67 @@
...
+	name = "Medic"
+
+/obj/item/storage/box/exploring/medic/PopulateContents()
+	new /obj/item/clothing/suit/space/eva(src)
+	new /obj/item/clothing/head/helmet/space/eva(src)
+	new /obj/item/gun/energy/kinetic_accelerator(src)
+	new /obj/item/storage/belt/medical(src)
+	new /obj/item/reagent_containers/medspray/silver_sulf(src)
+	new /obj/item/reagent_containers/medspray/styptic(src)
+	new /obj/item/stack/medical/gauze
```
Comment (Archanial) [inline code/game/objects/items/miscellaneous.dm:186]:
```suggestion
	new /obj/item/stack/medical/gauze(src)
```

File: code/modules/jobs/job_types/explorer.dm [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+	total_positions = 3
+	spawn_positions = 3
+	supervisors = "the quartermaster and the head of personnel"
+	selection_color = "#dcba97"
+	chat_color = "#CE957E"
+
+	outfit = /datum/outfit/job/explorer
+
+	access = list(ACCESS_MAINT_TUNNELS, ACCESS_MAILSORTING, ACCESS_CARGO, ACCESS_QM, ACCESS_MINING, ACCESS_MECH_MINING, ACCESS_MINING_STATION, ACCESS_MINERAL_STOREROOM, ACCESS_GATEWAY, ACCESS_EVA)
+	minimal_access = list(ACCESS_MINING, ACCESS_MECH_MINING, ACCESS_MINING_STATION, ACCESS_MAILSORTING, ACCESS_MINERAL_STOREROOM, ACCESS_GATEWAY, ACCESS_EVA, ACCESS_MAINT_TUNNELS)
```
Comment (Archanial) [inline code/modules/jobs/job_types/explorer.dm:16]:
```suggestion
	minimal_access = list(ACCESS_MINING, ACCESS_MECH_MINING, ACCESS_MINING_STATION, ACCESS_MAILSORTING, ACCESS_MINERAL_STOREROOM, ACCESS_GATEWAY, ACCESS_EVA)
```
No need for maints, it's not part of his job

**Review by Archanial — COMMENTED** [review]
Still don't really like the idea of splitting mining and exploring.
Nothing blocks miners from exploring and explorer from mining, they are the same roles but explorer gets more stuff for some reason.

**Comment by Fronsis** [comment]
I would personally love the idea of Science using the science outpost on lavaland to get research point/some new toys to experiment with the help of miners, that could lead to adding more mobs to the fauna and some new ruins, on top of that if on parallel you get space ruins as a job like Caecilius pr tried to do with Gateway and Bacon's Space exploration and you've got the perfect formula

**Comment by That0nePerson** [comment]
I feel like this role is already completely covered by curator. Also, them not having a room feels completely off.

**Comment by PowerfulBacon** [comment]
aka assistant

**Comment by PowerfulBacon** [comment]
Keep in mind this role is useless 70% of the time since our ruin budget is low meaning there is nothing in space.
You can explore the entirety of space in about 20 minutes with a jetpack and a fuel canister.
This job is literally the purpose of the curator.
This job is pretty bad for assassination objectives.

**Comment by OceanFish1** [comment]
Yeah but why is it cargo? This actually makes them a worse miners with loot, that only sometimes get better than mining loot if we forget about megafauna. With megafauna miners still get better loot.

**Comment by KazooBard** [comment]
> isn't miner just a better version of this job after they've gotten their lavaland loot?

No. There is no better version, as the loot they get differs and mostly since the explorer brings technology useful for science such as the slime exctracts and even syndicate gear and the miner brings artifacts that are usually combat based. There is a line between them. The loot they gather has different purposes, the rare bananium is found in many places in space, but just one on lavaland.

**Comment by ike709** [comment]
This role is useless, ruins are practically nonexistent due to memory issues and away missions IIRC are disabled.

**Comment by eeSPee** [comment]
I like the idea, but not the execution.

First things first, this does not function if asteroids do not spawn on each station/asteroid mining is not a thing, because the space is highly random and it's very unlikely you will get any worthwhile loot.

Second, the starting gear. All of the gear gives him EVA, but would this not be more easily accomplished by giving him EVA access? Some kit give him access to mining gear (PKA, for example), but keep in mind the Auxiliary Base already has a spare locker with miner's gear... Why not give him access to aux base like all the miners?  

Third, the beacon seems to overpowered/powergamey, and gives way too much loot.

>Security

Dual PKA? Why on earth? Miners don't get two, so why would the Explorer do. Why is there a security pack in the first place? Not like this dude will go around the station hunting crooks. Keep in mind PKAs deal a lot of damage in space.

>Medic

Fluton and health sunglasses? Why? A red medkit should be more than enough for your space exploration needs. 

>Field Engineer

Engineers already have space suits, space access and better gear. This pack is very underwhelming.

But the biggest flaw with this is that doesn't add anything new, honestly. This is no different from the miner, except it gets a super overpowered delivery pack. Even more so, a miner can get all the gear this guy has and do what this guy is doing; get a PKA, a space suit from a pod, a few supplies from medbay, and you're good to go.

**Comment by KazooBard** [comment]
> Keep in mind this role is useless 70% of the time since our ruin budget is low meaning there is nothing in space.
> You can explore the entirety of space in about 20 minutes with a jetpack and a fuel canister.
> This job is literally the purpose of the curator.
> This job is pretty bad for assassination objectives.

Sorry for late replies, I was unaware of lowered ruin spawn rates which does in fact lower the efficiency and potential of the role. I still do not agree with 20 minute space exploration.

**Comment by KazooBard** [comment]
@eeSPee 
> why the eva access?
Well, the explorer was implemented on a smaller station with actual players, which led to complaints about eva being gone quickly and there not being enough eva for crew during emergencies.

> powergamey
I was hoping to get feedback from people on what they think about the current state of this role, yet balancing proves to br the hardest part, as everything will be either too good or not good enough.

> engineer is underwhelming

in comparison, yes. The damaged rcd feels overpowred yet compared to other kits an engineer without it feels simply lackluster. 

What do you think about adding an upgraded welding tool instead? 

> medic can do just with advanced kit
understandable, the fulton was meant to make it easy for transporting heavily wounded exploring staff to the station from other z levels, os that they don’t get the lavaland gib experience.

> two pkas is too much
Now, the security kit’s idea was for experienced players that do not need many tools or meds to succesfuly raid ruins to get an edge by choosing offense over utility. The medkit is regular so it cannot be used in depressurized enviroments.

**Comment by PowerfulBacon** [comment]
> 
> 
> This role is useless, ruins are practically nonexistent due to memory issues and away missions IIRC are disabled.

This exactly. The only way I can see this working is if you incorporate it into the Curator since their job is based on exploring and already get some kits to help.
If you can make it around this then feel free to reopen, but as a new job I am going to say it probably wont work.

**Comment by KazooBard** [comment]
> > This role is useless, ruins are practically nonexistent due to memory issues and away missions IIRC are disabled.
> 
> This exactly. The only way I can see this working is if you incorporate it into the Curator since their job is based on exploring and already get some kits to help.
> If you can make it around this then feel free to reopen, but as a new job I am going to say it probably wont work.

As in rework the space exploration kit for curator by splitting it into specialisations?

**Pull request closed without merging**